packages/next/src/client/components/router-reducer/ppr-navigations.ts

```diff
@@ -355,8 +354,8 @@ export function listenForDynamicRequest(
 ) {
   responsePromise.then(
     ({ flightData }: FetchServerResponseResult) => {
-      for (const flightDataPath of flightData) {
-        if (typeof flightDataPath === 'string') {
+      for (const normalizedFlightData of flightData) {
+        if (typeof normalizedFlightData === 'string') {
```

Comment:
In doing this refactor I realized the original logic here seems wrong since `flightData` would be a string, not `flightDataPath`. It's likely this branch is never being hit in any tests, but I plan to follow-up on it shortly.